From: Gilles Talis <gilles.talis@gmail.com>

On PPC, if __NO_FPRS__ is defined, fparg_count and NUM_FPR_ARG_REGISTERS
are not defined, thus must not be used.

Signed-off-by: Gilles Talis <gilles.talis@gmail.com>
---
 src/powerpc/ffi.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/powerpc/ffi.c b/src/powerpc/ffi.c
index f3a96a1..9c69584 100644
--- a/src/powerpc/ffi.c
+++ b/src/powerpc/ffi.c
@@ -376,9 +376,10 @@ ffi_prep_args_SYSV (extended_cif *ecif, unsigned *const stack)
      with the number found in ffi_prep_cif_machdep().  However, intarg_count
      is incremeneted whenever we place an FP arg on the stack, so account for
      that before our assert test.  */
+#ifndef __NO_FPRS__
   if (fparg_count > NUM_FPR_ARG_REGISTERS)
     intarg_count -= fparg_count - NUM_FPR_ARG_REGISTERS;
-#ifndef __NO_FPRS__
+
   FFI_ASSERT (fpr_base.u
 	      <= stacktop.u - ASM_NEEDS_REGISTERS - NUM_GPR_ARG_REGISTERS);
 #endif
-- 
1.7.10.4

